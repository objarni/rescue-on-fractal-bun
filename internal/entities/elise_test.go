package entities

import (
	"fmt"
	approvals "github.com/approvals/go-approval-tests"
	px "github.com/faiface/pixel"
	"objarni/rescue-on-fractal-bun/internal/events"
	"objarni/rescue-on-fractal-bun/tests"
	"strings"
	"testing"
)

/*
- #initial state
- walking right
- walking left
- jumping when standing still
- jumping when pressing right
- clicking action when standing still
- clicking action when mid-air
- taking damage
*/

func Test_eliseInitial(t *testing.T) {
	elise := MakeElise(px.V(10, 20))
	var state = printElise(elise)
	approvals.VerifyString(t, state)
}

func Test_pressingLeft(t *testing.T) {
	elise := MakeElise(px.V(10, 20))
	elise = elise.Handle(EventBox{
		Event: events.KeyLeftDown,
		Box:   px.Rect{},
	})
	elise = elise.Tick(0, nil)
	var state = printElise(elise)
	approvals.VerifyString(t, state)
}

func printElise(elise Entity) string {
	state := fmt.Sprintf("Elise %v", "standing")
	hb := fmt.Sprintf("HitBox %v", printRect(elise.HitBox()))
	facing := fmt.Sprintf("Facing right")
	all := []string{state, hb, facing}
	return strings.Join(all, "\n") + "\n"
}

func printRect(box px.Rect) interface{} {
	return fmt.Sprintf("[%1.0f,%1.0f->%1.0f,%1.0f]",
		box.Min.X, box.Min.Y, box.Max.X, box.Max.Y)
}

func init() {
	approvals.UseReporter(tests.ReportWithMeld())
}
