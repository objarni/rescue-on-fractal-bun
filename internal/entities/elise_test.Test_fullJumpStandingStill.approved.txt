*** Scenario ***
* Events:
KeyJumpDown [-100000,-100000->100000,100000]

Ground height: 0
Tick count: 20

* Elise start state:
Elise standing
HitBox [-10,0->10,100]
Pos: <0.0,0.0>
Vel: <0.0,0.0>
Facing right

* Elise end state:
Elise jumping
HitBox [-10,2009->10,2109]
Pos: <0.0,2009.0>
Vel: <0.0,98.0>
Facing right


*** Simulation ***

 * Tick 1 *
Entity 0 is at [-10,0->10,100]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 2 *
Entity 0 is at [-10,130->10,230]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 3 *
Entity 0 is at [-10,230->10,330]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 4 *
Entity 0 is at [-10,329->10,429]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 5 *
Entity 0 is at [-10,429->10,529]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 6 *
Entity 0 is at [-10,529->10,629]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 7 *
Entity 0 is at [-10,628->10,728]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 8 *
Entity 0 is at [-10,727->10,827]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 9 *
Entity 0 is at [-10,826->10,926]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 10 *
Entity 0 is at [-10,926->10,1026]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 11 *
Entity 0 is at [-10,1025->10,1125]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 12 *
Entity 0 is at [-10,1123->10,1223]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 13 *
Entity 0 is at [-10,1222->10,1322]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 14 *
Entity 0 is at [-10,1321->10,1421]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 15 *
Entity 0 is at [-10,1420->10,1520]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 16 *
Entity 0 is at [-10,1518->10,1618]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 17 *
Entity 0 is at [-10,1616->10,1716]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 18 *
Entity 0 is at [-10,1715->10,1815]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 19 *
Entity 0 is at [-10,1813->10,1913]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

 * Tick 20 *
Entity 0 is at [-10,1911->10,2011]
Event: Wall [-1000,-1000->1000,0]
Event: KeyJumpDown [-100000,-100000->100000,100000]
Consequences:
- Entity 0 needs to handle KeyJumpDown

